server/src/main/kotlin/fi/oph/ludos/assignment/AssignmentRepository.kt

```diff
@@ -115,14 +111,31 @@ class AssignmentRepository(private val jdbcTemplate: JdbcTemplate) {
 
                 args.addAll(values)
             }
-            //            if (filters.topic != null) {
-            //                query += " AND assignment_topic ?"
-            //                args.add("%${filters.topic}%")
-            //            }
-            if (filters.language != null) {
-                query1 += " AND language = ?"
-                args.add(filters.language)
+
+            if (filters.oppimaara != null) {
+                println("assignmentTypeKoodiArvo ${filters.oppimaara}")
+                val values = filters.oppimaara.split(",")
+                val placeholders = values.joinToString(",") { "?" }
+
+                query1 += if (args.size > 0) {
+                    String.format(" AND suko_oppimaara_koodi_arvo IN (%s)", placeholders)
+                } else {
+                    String.format(" WHERE suko_oppimaara_koodi_arvo IN (%s)", placeholders)
+                }
+
+                args.addAll(values)
             }
+
+            if (filters.aihe != null) {
+                val arrayValues = filters.aihe.split(",").joinToString(", ") { "'$it'" }
+
+                query1 += if (args.size > 0) {
+                    String.format(" AND ARRAY[%s] <@ suko_aihe_koodi_arvos", arrayValues)
+                } else {
+                    String.format(" WHERE ARRAY[%s] <@ suko_aihe_koodi_arvos", arrayValues)
+                }
+            }
```

Comment:
Voiskoha tän tehdä ilman string.format?

Tyyliin:

 if (filters.aihe != null) {
                val arrayValues = filters.aihe.split(",").joinToString(", ") { "'$it'" }

                query1 += if (args.size > 0) {
                    " AND ARRAY[?] <@ suko_aihe_koodi_arvos"
                } else {
                    " WHERE ARRAY[?] <@ suko_aihe_koodi_arvos"
                }
                
                args.add(tähän jotenki argsit oikein formatoituna)
    }